src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -0,0 +1,390 @@
+        repair_type_overrides.get(repairType).repair_dc_groups = dcGroups;
+    }
+
+    public Set<String> getDCGroups(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.repair_dc_groups);
+    }
+
+    public boolean getRepairPrimaryTokenRangeOnly(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.repair_primary_token_range_only);
+    }
+
+    public void setRepairPrimaryTokenRangeOnly(RepairType repairType, boolean primaryTokenRangeOnly)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).repair_primary_token_range_only = primaryTokenRangeOnly;
+    }
+
+    public int getParallelRepairPercentageInGroup(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.parallel_repair_percentage_in_group);
+    }
+
+    public void setParallelRepairPercentageInGroup(RepairType repairType, int percentageInGroup)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).parallel_repair_percentage_in_group = percentageInGroup;
+    }
+
+    public int getParallelRepairCountInGroup(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.parallel_repair_count_in_group);
+    }
+
+    public void setParallelRepairCountInGroup(RepairType repairType, int countInGroup)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).parallel_repair_count_in_group = countInGroup;
+    }
+
+    public boolean getMVRepairEnabled(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.mv_repair_enabled);
+    }
+
+    public void setMVRepairEnabled(RepairType repairType, boolean enabled)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).mv_repair_enabled = enabled;
+    }
+
+    public void setForceRepairNewNode(RepairType repairType, boolean forceRepairNewNode)
+    {
+        ensureOverrides(repairType);
+        repair_type_overrides.get(repairType).force_repair_new_node = forceRepairNewNode;
+    }
+
+    public boolean getForceRepairNewNode(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.force_repair_new_node);
+    }
+
+    public String getTokenRangeSplitter(RepairType repairType)
+    {
+        return applyOverrides(repairType, opt -> opt.token_range_splitter);
+    }
+
+    // Options configures auto-repair behavior for a given repair type.
+    // All fields can be modified dynamically.
+    public static class Options implements Serializable
+    {
+        // The separator separating different DCs in repair_dc_groups
+        public static final String DC_GROUP_SEPARATOR = "\\|";
+
+        // defaultOptions defines the default auto-repair behavior when no overrides are defined
+        @VisibleForTesting
+        protected static final Options defaultOptions = getDefaultOptions();
+
+        public Options()
```

Comment:
Suggestion: implement `toString()` to return all the options like a map, otherwise `system_views.settings` can't read these:

```
                                auto_repair.repair_type_overrides | {incremental=org.apache.cassandra.repair.autorepair.AutoRepairConfig$Options@64af73a1, preview_repaired=org.apache.cassandra.repair.autorepair.AutoRepairConfig$Options@2ae56cae}
```

Reply:
Done